src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -475,6 +481,15 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     return device && device.id ? device : null;
   }
 
+  private isGroudpIdSame(groupId: string, kind: string, device: Device): boolean {
+    if (device === null) return false;
+    if (typeof device !== 'string')
+      device = JSON.parse(JSON.stringify(device)).deviceId
```

Comment:
Could we move JSON.stringify to a variable?

Reply:
done